src/Cogworks.CogFlare.Core/ViewComponents/CacheHeadersViewComponent.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Umbraco.Cms.Core.Models.Blocks;
+using UmbracoConstants = Umbraco.Cms.Core.Constants.PropertyEditors.Aliases;
+
+namespace Cogworks.CogFlare.Core.ViewComponents;
+
+public class CacheHeadersViewComponent : ViewComponent
+{
+    private readonly IUmbracoContextAccessor _umbracoContextAccessor;
+    private readonly CogFlareSettings _cogFlareSettings;
+
+    public CacheHeadersViewComponent(IUmbracoContextAccessor umbracoContextAccessor, CogFlareSettings cogFlareSettings)
+    {
+        _umbracoContextAccessor = umbracoContextAccessor;
+        _cogFlareSettings = cogFlareSettings;
+    }
+
+    public IViewComponentResult Invoke()
+    {
+        if (IsCacheable())
+        {
+            HttpContext.Response.Headers["Cache-Control"] = $"public, max-age={TimeConstants.OneMonth}";
+            HttpContext.Response.Headers["Edge-Control"] = $"cache-maxage={TimeConstants.OneMonth}s";
+        }
+        else
+        {
+            HttpContext.Response.Headers["Cache-Control"] = "private, no-cache, must-revalidate";
+        }
+
+        return Content("");
```

Comment:
You can ignore this but we are always returning string.Empty rather than "" in our VCs so we might just want to stick to one approach.